### PR TITLE
Issue#534 change modal conditionality

### PR DESCRIPTION
**Pull request by anthonypetersen**

### CODE REVIEW

File: src/events.js [hunk 1]
```diff
@@ -1925,13 +1925,49 @@ export const addEventSpecimenLinkForm = (formData) => {
 
     if (document.getElementById('navBarParticipantCheckIn')) document.getElementById('navBarParticipantCheckIn').dataset.connectId = connectId;
 
-    form.addEventListener('submit', e => {
+    form.addEventListener('submit', async (e) => {
         e.preventDefault();
-        btnsClicked(connectId, formData);
+        const collections = await getCollectionsByVisitType(connectId, '266600170');
```
Comment (anthonypetersen) [inline src/events.js:1930]: Would prefer this to be modular instead of hardcoding baseline collection CID

File: src/shared.js [hunk 2]
```diff
@@ -2015,4 +2015,18 @@ export const translateNumToType = {
   "0052": "NA",
   "0053": "NA",
   "0054": "NA"
-};
\ No newline at end of file
+};
+
+export const getCollectionsByVisitType = async (connectId, visitType) => {
```
Comment (anthonypetersen) [inline src/shared.js:2020]: There should already be a function you can use for getting collections by visit type

**Review by anthonypetersen — COMMENTED** [review]
Comments added

File: src/events.js (hunk 1 above)
Comment (geniosys) [reply src/events.js]: Resolved
https://github.com/episphere/biospecimen/pull/294/commits/43dd3db732689c038bf4ada84db6402063437c5b

File: src/shared.js (hunk 2 above)
Comment (geniosys) [reply src/shared.js]: resolved
https://github.com/episphere/biospecimen/pull/294/commits/981ccf6773cb6fa715996a70cbc012b9a7ba035e

**Pull request merged**